homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -147,80 +147,85 @@ def __init__(
-                self._state.advertised_time_at_location.astimezone(self._timezone)
-            )
-            if self._state.time_at_location:
-                self._attr_native_value = self._state.time_at_location.astimezone(
-                    self._timezone
-                )
-            if self._state.estimated_time_at_location:
-                self._attr_native_value = (
-                    self._state.estimated_time_at_location.astimezone(self._timezone)
-                )
+        self._attr_available = True
 
-            self._attr_extra_state_attributes[
-                ATTR_DEPARTURE_STATE
-            ] = self._state.get_state().name
-            self._attr_extra_state_attributes[ATTR_CANCELED] = self._state.canceled
+        self._attr_native_value = self._state.advertised_time_at_location.astimezone(
+            self._timezone
+        )
+        if self._state.time_at_location:
+            self._attr_native_value = self._state.time_at_location.astimezone(
+                self._timezone
+            )
+        if self._state.estimated_time_at_location:
+            self._attr_native_value = self._state.estimated_time_at_location.astimezone(
+                self._timezone
+            )
 
-            if self._state.other_information:
-                self._attr_extra_state_attributes[ATTR_OTHER_INFORMATION] = ", ".join(
-                    self._state.other_information
-                )
-            if self._state.deviations:
-                self._attr_extra_state_attributes[ATTR_DEVIATIONS] = ", ".join(
-                    self._state.deviations
-                )
-            self._delay_in_minutes = self._state.get_delay_time()
-            if self._delay_in_minutes:
-                self._attr_extra_state_attributes[ATTR_DELAY_TIME] = (
-                    self._delay_in_minutes.total_seconds() / 60
-                )
-            if self._state.advertised_time_at_location:
-                self._attr_extra_state_attributes[ATTR_PLANNED_TIME] = as_utc(
-                    self._state.advertised_time_at_location.astimezone(self._timezone)
-                ).isoformat()
-            if self._state.estimated_time_at_location:
-                self._attr_extra_state_attributes[ATTR_ESTIMATED_TIME] = as_utc(
-                    self._state.estimated_time_at_location.astimezone(self._timezone)
-                ).isoformat()
-            if self._state.time_at_location:
-                self._attr_extra_state_attributes[ATTR_ACTUAL_TIME] = as_utc(
-                    self._state.time_at_location.astimezone(self._timezone)
-                ).isoformat()
+        self._attr_extra_state_attributes[ATTR_DEPARTURE_STATE] = (
+            self._state.get_state().name if self._state.get_state().name else None
+        )
+        self._attr_extra_state_attributes[ATTR_CANCELED] = (
+            self._state.canceled if self._state.canceled else None
+        )
+        self._delay_in_minutes = self._state.get_delay_time()
```

Comment:
There is no need to use an instance variable here. You can use a local variable `delay_in_minutes`